src/utils/artifact/generateAragonArtifact.ts

```diff
@@ -0,0 +1,106 @@
+import { AragonAppJson, AragonArtifact, AragonEnvironment } from '~/src/types'
+import { keccak256, AbiItem } from 'web3-utils'
+import { getAppId } from '../appName'
+import { parseContractFunctions, AragonContractFunction } from '../ast'
+import { keyBy } from 'lodash'
+import { ethers } from 'ethers'
+
+const abiFallback = {
+  payable: true,
+  stateMutability: 'payable',
+  type: 'fallback'
+}
+
+function _generateAragonArtifact(
+  arapp: AragonAppJson,
+  abi: AbiItem[],
+  functions: AragonContractFunction[]
+): AragonArtifact {
+  const abiFunctions = abi.filter(abiElem => abiElem.type === 'function')
+  const abiBySignature = keyBy(abiFunctions, ethers.utils.formatSignature)
+
+  // Get mainnet env
+  const mainnetEnv: AragonEnvironment | undefined =
+    arapp.environments['mainnet']
+  const appName = (mainnetEnv || {}).appName || ''
+
+  return {
+    ...arapp,
+
+    // Artifact appears to require the abi of each function
+    functions: functions.map(parsedFn => ({
+      roles: parsedFn.roles.map(role => role.id),
+      notice: parsedFn.notice,
+      abi:
+        abiBySignature[parsedFn.sig] ||
+        (parsedFn.sig === 'fallback' ? abiFallback : null),
+      // #### Todo: Is the signature actually necessary?
+      // > Will keep them for know just in case, they are found in current release
+      sig: parsedFn.sig
+    })),
+
+    deprecatedFunctions: {},
+
+    // Artifact appears to require the roleId to have bytes precomputed
+    roles: (arapp.roles || []).map(role => ({
+      ...role,
+      bytes: keccak256(role.id)
+    })),
+
+    abi,
+    // Additional metadata
+    flattenedCode: './code.sol',
+    appName,
+    appId: appName ? getAppId(appName) : ''
```

Comment:
These values are available in a specific environment. We should not put this information here otherwise the ipfs hash will be different on each environment. We have to make sure the logic that depends on these bits handles the change accordingly.

Reply:
Added in https://github.com/aragon/buidler-aragon/issues/77